app/actions/remote/scheduled_post.test.ts

```diff
@@ -0,0 +1,125 @@
+import {getConfigValue} from '@queries/servers/system';
+
+import {deleteScheduledPost, fetchScheduledPosts} from './scheduled_post';
+
+import type ServerDataOperator from '@database/operator/server_data_operator';
+
+const serverUrl = 'baseHandler.test.com';
+
+const scheduledPostsResponse: FetchScheduledPostsResponse = {
+    directChannels: [],
+    bar: [
+        {
+            id: 'scheduled_post_id',
+            channel_id: 'channel_id',
+            root_id: '',
+            message: 'test scheduled post',
+            scheduled_at: 123,
+            user_id: 'user_id',
+            processed_at: 0,
+            update_at: 456,
+            error_code: '',
+        },
+        {
+            id: 'scheduled_post_id_2',
+            channel_id: 'channel_id',
+            root_id: '',
+            message: 'test scheduled post 2',
+            scheduled_at: 123,
+            user_id: 'user_id',
+            processed_at: 0,
+            update_at: 456,
+            error_code: '',
+        },
+    ],
+};
+
+const throwFunc = () => {
+    throw Error('error');
+};
+
+const mockClient = {
+    getScheduledPostsForTeam: jest.fn(() => Promise.resolve({...scheduledPostsResponse})),
+    deleteScheduledPost: jest.fn((scheduledPostId) => {
+        return Promise.resolve(scheduledPostsResponse.bar.find((post) => post.id === scheduledPostId));
+    }),
+};
+jest.mock('@queries/servers/system', () => ({
+    getConfigValue: jest.fn(),
+}));
+
+const mockedGetConfigValue = jest.mocked(getConfigValue);
+let operator: ServerDataOperator;
+
+beforeAll(() => {
+    // eslint-disable-next-line
+    // @ts-ignore
+    NetworkManager.getClient = () => mockClient;
+});
+
+beforeEach(async () => {
+    await DatabaseManager.init([serverUrl]);
+    operator = DatabaseManager.getServerDatabaseAndOperator(serverUrl).operator;
+});
+
+afterEach(async () => {
+    await DatabaseManager.destroyServerDatabase(serverUrl);
+});
+
+describe('fetchScheduledPosts', () => {
+    it('fetch Schedule post - handle database not found', async () => {
+        const result = await fetchScheduledPosts('foo', 'bar');
+        expect(result).toBeDefined();
+        expect(result.error).toBeTruthy();
+    });
+
+    it('fetch Schedule post - handle client error', async () => {
+        jest.spyOn(NetworkManager, 'getClient').mockImplementationOnce(throwFunc);
+
+        const result = await fetchScheduledPosts(serverUrl, 'bar');
+        expect(result).toBeDefined();
+        expect(result.error).toBeTruthy();
+    });
+
+    it('fetch Schedule post - handle scheduled post disabled', async () => {
+        const result = await fetchScheduledPosts(serverUrl, 'bar');
+        expect(result).toBeDefined();
+        expect(result.scheduledPosts).toEqual([]);
+    });
+
+    it('fetch Schedule post - handle scheduled post enabled', async () => {
+        mockedGetConfigValue.mockResolvedValueOnce('true');
+        jest.spyOn(operator, 'handleScheduledPosts').mockResolvedValueOnce([]);
```

Comment:
Should you mock this or should you check that handleScheduledPosts() .toHaveBeenCalledWith() properly?

Reply:
Done.